### PR TITLE
Clear cache and refresh user profile on email verification

### PR DESCRIPTION
**Pull request by DavisRayM**
### Changes / Features implemented

- Refresh user profile from database before updating metadata
- Clear user-profiles cache on email verification

### Steps taken to verify this change does what is intended

- N/A

### Side effects of implementing this change

- N/A

Possible fix for #1948

### CODE REVIEW

**Review by FrankApiyo — APPROVED** [review]

**Review by ivermac — DISMISSED** [review]

File: onadata/apps/api/viewsets/user_profile_viewset.py [hunk 1]
```diff
@@ -100,7 +100,8 @@ def serializer_from_settings():
 
 
 def set_is_email_verified(profile, is_email_verified):
-    profile.metadata.update({"is_email_verified": is_email_verified})
+    profile.refresh_from_db()
```
Comment (ukanga) [inline onadata/apps/api/viewsets/user_profile_viewset.py:103]: Can we make this call before we call `set_is_email_verified()`? I see a reason why the `refresh_from_db` may need to happen on the GET request to verify, but perhaps not on the POST request. I would hope all requests on a POST use the master DB. 

Is this issue due to a read replica update delay?

Comment (DavisRayM) [reply onadata/apps/api/viewsets/user_profile_viewset.py:103]: Yes, I can change this.

Comment (DavisRayM) [reply onadata/apps/api/viewsets/user_profile_viewset.py:103]: Doesn't seem to be a read replica update delay; the change never shows up. Noticed accounts that had been verified a while back(2-3 days) but the metadata wasn't updated

Comment (ukanga) [reply onadata/apps/api/viewsets/user_profile_viewset.py:103]: What then guarantees this issue is resolve considering the statement has always been there? The only new statements is the cache deletion and the refresh_from_db.

Comment (DavisRayM) [reply onadata/apps/api/viewsets/user_profile_viewset.py]: Changed this in the latest commits. The main issue seems to have been the `RegistrationProfile` retrieval in the user profile viewset.

**Review by ivermac — DISMISSED** [review]

**Review by ivermac — APPROVED** [review]

**Review by ukanga — APPROVED** [review]

**Pull request merged**